python/lsst/sconsUtils/state.py

```diff
@@ -128,6 +130,11 @@ def _initEnvironment():
                 ourEnv[varname] = os.environ[varname]
                 ourEnv[k] = os.environ[k]
 
+    # add <build root>/ups directory to the configuration search path
+    # this allows the .cfg file for the package being built to be found without
+    # requiring <product name>_DIR to be in the env
+    cfgPath.append(os.path.join(SCons.Script.Dir('#').abspath, 'ups'))
```

Comment:
I'm not clear as to why the value of all env vars postfixed with `_DIR` are pulled into `cfgPath`.  Does anyone know? Is it possible that the .cfg/.table file for the product being built are located outside of its source tree?